static/app/components/platformPicker.tsx

```diff
@@ -82,6 +82,12 @@ class PlatformPicker extends Component<PlatformPickerProps, State> {
     filter: this.props.noAutoFilter ? '' : (this.props.platform || '').split('-')[0]!,
   };
 
+  componentDidUpdate(prevProps: Readonly<PlatformPickerProps>): void {
+    if (this.props.defaultCategory !== prevProps.defaultCategory) {
+      this.setState({category: this.props.defaultCategory ?? categoryList[0]!.id});
+    }
+  }
```

Comment:
This is needed because when navigating back, we want to display the previously selected platform. If the selected platform isn't part of the "Popular" tab, it won't be visible by default, which can confuse users (as it was initially to me).